### PR TITLE
fix: catch UnicodeDecodeError in parse_events, _read_config_model, and get_all_sessions (#369)

### LINKED ISSUE #369: [aw][code health] UnicodeDecodeError propagates uncaught from parse_events and _read_config_model
## Root Cause

Two paths in the parser crash the tool (instead of skipping gracefully) when UTF-8 decoding fails:

**`parse_events` (`parser.py`)**

The file is opened with `encoding="utf-8"`, but the per-line iteration is only protected against `json.JSONDecodeError` and pydantic `ValidationError`. A `UnicodeDecodeError` raised by Python's text-mode file iteration (which happens before `json.loads` is reached) propagates uncaught:

```python
with events_path.open(encoding="utf-8") as fh:
    for lineno, line in enumerate(fh, start=1):   # ← UnicodeDecodeError raised here
        ...
        try:
            raw = json.loads(stripped)
        except json.JSONDecodeError:               # ← only JSON errors caught
            ...
```

`get_all_sessions` only catches `OSError`, so a corrupt event file crashes the entire session listing.

**`_read_config_model` (`parser.py`)**

```python
data = json.loads(path.read_text(encoding="utf-8"))
#                 ^^^^^^^^^^^^ raises UnicodeDecodeError if file has non-UTF-8 bytes
```

The `except` clause covers `json.JSONDecodeError` and `OSError`, but not `UnicodeDecodeError`.

## Impact

- Any `events.jsonl` file containing a line with invalid UTF-8 bytes (e.g. injected binary content from a tool response) raises an unhandled exception in `parse_events`, which propagates through `get_all_sessions` and crashes `copilot-usage summary`/`cost`/`live` entirely rather than skipping the affected session.
- Malformed `~/.copilot/config.json` with invalid UTF-8 bytes crashes `build_session_summary` for any active session where model detection falls back to the config file.

Note: the existing `corrupt-session` fixture only covers malformed JSON, not invalid UTF-8 encoding.

### Fix

1. **`parse_events`** — catch `UnicodeDecodeError` around the line-read loop (or at the `open()` call) and log a warning + skip the affected file (consistent with how `get_all_sessions` handles `OSError`):
   ```python
   try:
       with events_path.open(encoding="utf-8") as fh:
           for lineno, line in enumerate(fh, start=1):
               ...
   except UnicodeDecodeError as exc:
       logger.warning("{} — UTF-8 decode error, skipping session: {}", events_path, exc)
       return events  # return what was parsed so far, or []
   ```
   Alternatively, open with `errors="replace"` and log a warning if any replacement character appears.

2. **`_read_config_model`** — add `UnicodeDecodeError` to the `OSError` except clause:
   ```python
   except (OSError, UnicodeDecodeError) as exc:
       logger.debug("Could not read config file {}: {}", path, exc)
       return None
   ```

3. **`get_all_sessions`** — broaden the except clause to cover `UnicodeDecodeError` as a belt-and-suspenders guard:
   ```python
   except (OSError, UnicodeDecodeError) as exc:
       logger.warning("Skipping unreadable session {}: {}", events_path, exc)
       continue
   ```

### Tests

- Add a test fixture (or `tmp_path`-based test) where `events.jsonl` contains a line with a raw invalid UTF-8 byte sequence. Assert that `parse_events` returns a partial list (skipping the bad line or the whole file gracefully) rather than raising.
- Add a test for `_read_config_model` given a config file with non-UTF-8 bytes — assert it returns `None` without raising.
- Regression: the existing corrupt-session tests should continue to pass unchanged.

> Generated by [Code Health Analysis](https://github.com/microsasa/cli-tools/actions/runs/23559889825) · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Fcode-health%22&type=issues)

### PR DESCRIPTION
**Pull request by microsasa**
Closes #369

## Problem

Three paths in `parser.py` crash the tool when UTF-8 decoding fails on corrupt session/config files:

1. **`parse_events`** — `UnicodeDecodeError` raised during line iteration propagates uncaught through `get_all_sessions`, crashing `copilot-usage summary`/`cost`/`live` entirely
2. **`_read_config_model`** — `path.read_text(encoding="utf-8")` raises `UnicodeDecodeError` not covered by the `except` clause
3. **`get_all_sessions`** — only catches `OSError`, so any `UnicodeDecodeError` from `parse_events` also crashes

## Fix

1. **`parse_events`** — wrap the file open/read loop in `try/except UnicodeDecodeError`, log a warning, and return events parsed so far
2. **`_read_config_model`** — add `UnicodeDecodeError` to the `OSError` except clause
3. **`get_all_sessions`** — broaden the except clause to `(OSError, UnicodeDecodeError)` as a belt-and-suspenders guard

## Tests added

- `test_unicode_decode_error_returns_partial` — `events.jsonl` with valid first line + invalid UTF-8 bytes returns partial list
- `test_unicode_decode_error_full_file` — entirely invalid UTF-8 `events.jsonl` returns empty list
- `test_unicode_decode_error_returns_none` — `config.json` with non-UTF-8 bytes returns `None`
- `test_unicode_decode_error_session_is_skipped` — `get_all_sessions` skips bad session, keeps good ones

All 678 tests pass, 99.36% coverage.

> Generated by [Issue Implementer](https://github.com/microsasa/cli-tools/actions/runs/23565717914) · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+%22gh-aw-workflow-id%3A+issue-implementer%22&type=pullrequests)

### CODE REVIEW

File: tests/copilot_usage/test_parser.py [hunk 1]
```diff
@@ -2778,6 +2806,39 @@ def _flaky_open(self: Path, *args: object, **kwargs: object) -> object:  # type:
...
+        d_bad.mkdir()
+        (d_bad / "events.jsonl").write_bytes(
+            b'{"type": "session.start"}\n\xff\xfe\x80\x81\n'
+        )
+
+        results = get_all_sessions(tmp_path)
+        # The good session should be present; the bad one skipped
+        session_ids = [r.session_id for r in results]
+        assert "sess-good" in session_ids
+
```
[the remark this answers is not in the document]
Comment (microsasa) [reply tests/copilot_usage/test_parser.py]: Added both assertions: `assert "sess-bad" not in session_ids` and `assert len(session_ids) == 1` to verify the bad session is actually skipped.

> Generated by [Review Responder](https://github.com/microsasa/cli-tools/actions/runs/23566053021)

> [!WARNING]
> <details>
> <summary>⚠️ Firewall blocked 1 domain</summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `astral.sh`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "astral.sh"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>

File: tests/copilot_usage/test_parser.py [hunk 2]
```diff
@@ -2778,6 +2810,41 @@ def _flaky_open(self: Path, *args: object, **kwargs: object) -> object:  # type:
...
+            )
+            + "\n",
+            encoding="utf-8",
+        )
+
+        # Create a session with invalid UTF-8 bytes
+        d_bad = tmp_path / "sess-bad"
+        d_bad.mkdir()
+        (d_bad / "events.jsonl").write_bytes(
+            b'{"type": "session.start"}\n\xff\xfe\x80\x81\n'
```
[the remark this answers is not in the document]
Comment (microsasa) [reply tests/copilot_usage/test_parser.py]: Changed the bad `events.jsonl` to be entirely invalid bytes (`\xff\xfe\x80\x81\n`) from the start, eliminating potential flakiness from buffered I/O.

> Generated by [Review Responder](https://github.com/microsasa/cli-tools/actions/runs/23566434554)
